### PR TITLE
Daniel he

### PR DESCRIPTION
**Pull request by student-danielhe**
Exploding enemy and projectile implemented

### CODE REVIEW

File: objects/obj_enemy_aggressive/Alarm_1.gml
Comment (CamoKnight117) [inline objects/obj_enemy_aggressive/Alarm_1.gml:1]: Need semi colons to follow style guide.

File: objects/obj_enemy_aggressive/Create_0.gml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/// @description Insert description here
+// You can write your code in this editor
+
+
+
+// Inherit the parent event
+event_inherited();
+
+attackRate=240
```
Comment (CamoKnight117) [inline objects/obj_enemy_aggressive/Create_0.gml:9]: ;

Comment (CamoKnight117) [inline objects/obj_enemy_aggressive/Create_0.gml:9]:
```suggestion
attackRate=240;
```

File: objects/obj_enemy_aggressive/Alarm_1.gml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/// @description Insert description here
+// You can write your code in this editor
+vspeed=-5
```
Comment (CamoKnight117) [inline objects/obj_enemy_aggressive/Alarm_1.gml:3]:
```suggestion
vspeed=-5;
```

Comment (CamoKnight117) [inline objects/obj_enemy_aggressive/Alarm_1.gml:3]:
```suggestion
vspeed=-5;
```

File: objects/obj_enemy_aggressive/Alarm_1.gml [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+alarm_set(1,attackRate)
```
Comment (CamoKnight117) [inline objects/obj_enemy_aggressive/Alarm_1.gml:4]:
```suggestion
alarm_set(1,attackRate);
```

File: objects/obj_enemy_aggressive/Step_0.gml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+
+// Inherit the parent event
+event_inherited();
+is_grounded = instance_place(x, y, [obj_ground_parent]);
+if (is_grounded <= 0) {
+	vspeed = vspeed+0.1;
+	is_falling = instance_place(x, y+vspeed, [obj_ground_parent]);
+	if (is_falling > 0) {
+		vspeed=0
```
Comment (CamoKnight117) [inline objects/obj_enemy_aggressive/Step_0.gml:13]:
```suggestion
		vspeed=0;
```

File: objects/obj_enemy_aggro_range/Create_0.gml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/// @description Insert description here
+// You can write your code in this editor
+
+
+
+// Inherit the parent event
+event_inherited();
+
+range=500
```
Comment (CamoKnight117) [inline objects/obj_enemy_aggro_range/Create_0.gml:9]:
```suggestion
range=500;
```

File: objects/obj_enemy_aggro_range/Create_0.gml [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+aggro=0
```
Comment (CamoKnight117) [inline objects/obj_enemy_aggro_range/Create_0.gml:10]:
```suggestion
aggro=0;
```

File: objects/obj_enemy_aggro_range/Step_0.gml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+/// @description Insert description here
+// You can write your code in this editor
+
+
+
+// Inherit the parent event
+event_inherited();
+  var player=collision_rectangle(x-range,y-range,x+range,y+range,[obj_player],0,0);
+  if player != noone
```
Comment (CamoKnight117) [inline objects/obj_enemy_aggro_range/Step_0.gml:9]:
```suggestion
  if (player != noone)
```

File: objects/obj_enemy_aggro_range/Step_0.gml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  {
+	aggro=true
```
Comment (CamoKnight117) [inline objects/obj_enemy_aggro_range/Step_0.gml:11]:
```suggestion
	aggro=true;
```

File: objects/obj_enemy_jump/Alarm_1.gml [hunk 9]
```diff
@@ -0,0 +1,5 @@
+/// @description Jump
+// You can write your code in this editor
+vspeed=-jumpHeight
+alarm_set(1,jumpRate)
```
Comment (CamoKnight117) [inline objects/obj_enemy_jump/Alarm_1.gml:4]:
```suggestion
alarm_set(1,jumpRate);
```

File: objects/obj_enemy_aggressive/Alarm_1.gml
Comment (CamoKnight117) [reply objects/obj_enemy_aggressive/Alarm_1.gml:1]: Also need comments on all events replacing "Insert description here" and deleting "You can write your code in this editor"

File: objects/obj_enemy_jump/Create_0.gml [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+
+
+// Inherit the parent event
+event_inherited();
+
+grav=0.2
+jumpRate=120
+jumpHeight=5
+alarm_set(1,jumpRate)
```
Comment (CamoKnight117) [inline objects/obj_enemy_jump/Create_0.gml:12]:
```suggestion
grav=0.2;
jumpRate=120;
jumpHeight=5;
alarm_set(1,jumpRate);
```

File: objects/obj_enemy_jump/Step_0.gml [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+
+// Inherit the parent event
+event_inherited();
+is_grounded = instance_place(x, y, [obj_ground_parent]);
+if (is_grounded <= 0) {
+	vspeed = vspeed+grav;
+	is_falling = instance_place(x, y+vspeed, [obj_ground_parent]);
+	if (is_falling > 0) {
+		vspeed=0
```
Comment (CamoKnight117) [inline objects/obj_enemy_jump/Step_0.gml:13]:
```suggestion
		vspeed=0;
```

File: objects/obj_enemy_projectile/Step_0.gml [hunk 12]
```diff
@@ -2,9 +2,9 @@
 
 y_spd -= grav;
 y -= y_spd;
-
+/*
 if (y > initial_y) {
 	instance_destroy(self);
-}
+}*/
```
Comment (CamoKnight117) [inline objects/obj_enemy_projectile/Step_0.gml:8]: Do we still need this code that's commented out? If we're not saving it for anything, just get rid of it

File: objects/obj_explode_projectile/Collision_obj_player_mask.gml [hunk 13]
```diff
@@ -0,0 +1,4 @@
+/// @description Insert description here
+// You can write your code in this editor
+instance_create_layer(x,y,"Instances",obj_exposion)
```
Comment (CamoKnight117) [inline objects/obj_explode_projectile/Collision_obj_player_mask.gml:3]:
```suggestion
instance_create_layer(x,y,"Instances",obj_exposion);
```

File: objects/obj_explode_projectile/Create_0.gml [hunk 14]
```diff
@@ -0,0 +1,3 @@
+/// @description Instantiate variables
+
+hspeed=-10
```
Comment (CamoKnight117) [inline objects/obj_explode_projectile/Create_0.gml:3]:
```suggestion
hspeed=-10;
```

File: objects/obj_exposion/Alarm_1.gml [hunk 15]
```diff
@@ -0,0 +1,4 @@
+/// @description Despawn
+// You can write your code in this editor
+
+instance_destroy()
```
Comment (CamoKnight117) [inline objects/obj_exposion/Alarm_1.gml:4]:
```suggestion
instance_destroy();
```

File: objects/obj_exposion/Create_0.gml [hunk 16]
```diff
@@ -0,0 +1,8 @@
+/// @description Set despawn timer
+// You can write your code in this editor
+
+
+
+// Inherit the parent event
+event_inherited();
+alarm_set(1,30)
```
Comment (CamoKnight117) [inline objects/obj_exposion/Create_0.gml:8]:
```suggestion
alarm_set(1,30);
```

File: objects/obj_hit_parent/Create_0.gml [hunk 17]
```diff
@@ -0,0 +1,4 @@
+/// @description Set damge for the hit
+// You can write your code in this editor
+dmg=10
```
Comment (CamoKnight117) [inline objects/obj_hit_parent/Create_0.gml:3]:
```suggestion
dmg=10;
```

**Comment by student-danielhe** [comment]
I think I fixed all the things we mentioned, please take a look again and help me merge it

**Review by CamoKnight117 — DISMISSED** [review]

**Review by CamoKnight117 — DISMISSED** [review]

**Review by CamoKnight117 — APPROVED** [review]

**Pull request merged**